File: Client/src/app/content/notes/full-note/models/api/photos/update-photos-collection-info-command.ts
import { BaseUpdateCollectionInfoCommand } from '../base-update-collection-info-command';

export class UpdatePhotosCollectionInfoCommand extends BaseUpdateCollectionInfoCommand {
  constructor(
    public noteId: string,
    public contentId: string,
    public name: string,
    public count: number,
    public width: string,
    public height: string,
  ) {
    super(noteId, contentId, name);
  }
}

File: Backend/NootsWEB/Noots.Editor/Commands/Photos/UpdatePhotosCollectionInfoCommand.cs
using System.ComponentModel.DataAnnotations;

namespace Noots.Editor.Commands.Photos;

public class UpdatePhotosCollectionInfoCommand : BaseUpdateCollectionInfo
{
    public UpdatePhotosCollectionInfoCommand(Guid noteId, Guid contentId, string name, string connectionId) : base(noteId, contentId, name, connectionId)
    {
    }

    [Range(1, 4)]
    public int Count { set; get; }

    [Required(AllowEmptyStrings = false)]
    public string Width { set; get; }

    [Required(AllowEmptyStrings = false)]
    public string Height { set; get; }
}
